internal/pkg/runtime/engine/singularity/container_linux.go

```diff
@@ -2069,17 +2074,84 @@ func (c *container) prepareNetworkSetup(system *mount.System, pid int) (func(con
 
 // addFuseMount transforms the plugin configuration into a series of
 // mount requests for FUSE filesystems
-func (c *container) addFuseMount(system *mount.System) error {
-	for i, name := range c.engine.EngineConfig.GetPluginFuseMounts() {
-		var cfg struct {
-			Fuse singularity.FuseInfo
+func (c *container) addFuseMount(system *mount.System) (int, error) {
+	fakeroot := c.engine.EngineConfig.GetFakeroot()
+	fakerootHybrid := fakeroot && os.Geteuid() != 0
+
+	uid := os.Getuid()
+	gid := os.Getgid()
+
+	// as fakeroot can change UID/GID, we allow others users
+	// to access FUSE mount point
+	allowOther := ""
+	if fakeroot {
+		allowOther = ",allow_other"
+	}
+	if fakerootHybrid {
+		uid = 0
+		gid = 0
+	}
+
+	fds := make([]int, 0)
+	usernsFd := -1
+
+	fuseMounts := c.engine.EngineConfig.GetFuseMount()
+
+	for _, fuseMount := range fuseMounts {
+		if fuseMount.FromContainer || !c.userNS {
+			continue
 		}
+		fds = append(fds, fuseMount.Fd)
+	}
 
-		if err := c.engine.EngineConfig.GetPluginConfig(name, &cfg); err != nil {
-			sylog.Debugf("Failed getting plugin config: %+v\n", err)
-			return err
+	if len(fds) > 0 {
+		socketPair := c.engine.EngineConfig.GetUnixSocketPair()
+
+		if err := c.rpcOps.SendFd(socketPair[1], fds); err != nil {
+			return usernsFd, fmt.Errorf("while requesting file descriptors send: %s", err)
+		}
+
+		buf := make([]byte, unix.CmsgSpace(len(fds)*4))
+		_, _, _, _, err := unix.Recvmsg(socketPair[0], nil, buf, 0)
+		if err != nil {
+			return usernsFd, fmt.Errorf("while receiving file descriptors: %s", err)
+		}
+
+		msgs, err := unix.ParseSocketControlMessage(buf)
+		if err != nil {
+			return usernsFd, fmt.Errorf("while parsing socket control message: %s", err)
+		}
+
+		newfds := make([]int, 0)
```

Comment:
should we make with size at least 1? If we don't get any FDs we won't do anything with this.... but otherwise we have at least 1 append.

Reply:
the check after will return appropriate error if there is no file descriptor returned